src/splitting.rs

```diff
@@ -47,6 +47,11 @@ impl<S: WordSplitter + ?Sized> WordSplitter for Box<S> {
     }
 }
 */
+impl<T: WordSplitter> WordSplitter for &T {
```

Comment:
This is another funny looking construct :-) Is this perhaps a use case for the [`Borrow`](https://doc.rust-lang.org/std/borrow/trait.Borrow.html#examples) stdlib trait?

What I mean is that the function that have a type bound on `WordSplitter` could perhaps use a bound which allows them to accept both a `WordSplitter` and a borrowed `WordSplitter`.

Reply:
It's a fairly common pattern - see [`AsRef`](https://doc.rust-lang.org/stable/std/convert/trait.AsRef.html#impl-AsRef%3CU%3E), [`Display`](https://doc.rust-lang.org/stable/std/fmt/trait.Display.html#impl-Display-82), [`Hash`](https://doc.rust-lang.org/stable/std/hash/trait.Hash.html#impl-Hash-59), [`Fn`](https://doc.rust-lang.org/stable/std/ops/trait.Fn.html#impl-Fn%3CA%3E). It's also required for the implementation of `From<&Options<'_, S>>` for `Options<'_, &S>` (notice it produces an `&S`, which allows `S` to not be `Clone`).

Reply:
Thanks for those references to the stdlib! This makes me feel like we're in good company :-)